src/d3a/models/market/blockchain_interface.py

```diff
@@ -71,121 +69,21 @@ def bc_listener(self):
         pass
 
 
-class SubstrateBlockchainInterface:
-    def __init__(self, bc):
-        self.contracts = {'main': main_address}
-        self.substrate = bc
+class SubstrateBlockchainInterface(BlockChainInterface):
+    def __init__(self):
+        super().__init__()
```

Comment:
I think you can completely omit the constructor, this is the default one.

Reply:
I will add to the blockchain ticket that deals with the offer functions in blockchain utils